pandas/core/array_algos/masked_reductions.py

```diff
@@ -100,3 +100,12 @@ def min(values: np.ndarray, mask: np.ndarray, skipna: bool = True):
 
 def max(values: np.ndarray, mask: np.ndarray, skipna: bool = True):
     return _minmax(np.max, values=values, mask=mask, skipna=skipna)
+
+
+def mean(values: np.ndarray, mask: np.ndarray, skipna: bool = True):
```

Comment:
can you type the result Union[Scalar libmissing.NA], though maybe that's best done in another pass.

Reply:
can you do this

Reply:
we need to support min_count here (just pass it thru to _sumprod)

Reply:
Let's leave the output types for another PR (none of the other functions here have output types, they all only are typed for the input)

> we need to support min_count here (just pass it thru to _sumprod)

`min_count` is not an argument for `mean` (check eg `Series.mean`, we only have the keyword for prod/sum)